src/test/ui/const-generics/issues/issue-67739.rs

```diff
@@ -10,7 +10,7 @@ pub trait Trait {
 
     fn associated_size(&self) -> usize {
         [0u8; mem::size_of::<Self::Associated>()];
```

Comment:
This ICEs the compiler, because CTFE tries to `subst_and_normalize_erasing_regions`, and this can't be normalized. I'm not sure how to resolve this. @eddyb a while back we worked on this and I had some hacky stuff that first substituted and returned `TooGeneric` if the result still had substs. Back then you had the insight that allowed the hack to be removed. I'm hoping this is just another case of that...

The entry point is https://github.com/rust-lang/rust/blob/8e6de3244cd62cbde367c206bc9972daded5281d/src/librustc_mir/interpret/operand.rs#L493-L496 which invokes https://github.com/rust-lang/rust/blob/7cdbc87a49b0b705a41a004a6d486b0952521ae7/src/librustc_mir/interpret/eval_context.rs#L338-L347 which ICEs in https://github.com/rust-lang/rust/blob/8e6de3244cd62cbde367c206bc9972daded5281d/src/librustc_traits/normalize_erasing_regions.rs#L13-L39

Reply:
Is codegen calling `self.monomorphize` correctly for the length?

Reply:
Codegen dosn't have this problem I think, as it would have substituted `Self` with the concrete type already.

Reply:
Judging by [the query stack](https://gist.github.com/oli-obk/6bf4cd07b030d536830f1b94cc8e84c9#file-gistfile1-txt-L636-L641) something is dropping a `ParamEnv` on the floor (`caller_bounds` should have a `Self: Trait` in it).

Reply:
that was it, thanks. It works now